src/main/java/Ryan.java

```diff
@@ -0,0 +1,308 @@
+        for (int i = 0; i < taskCounter; i++) {
+            System.out.println(tasks[i]);
+        }
+        printTaskNumber(taskCounter);
+    }
+
+    /**
+     * Prints the number of tasks
+     *
+     * @param taskCounter number of tasks in the array
+     */
+    private static void printTaskNumber(int taskCounter) {
+        String task = TASK_PLURAL;
+        if (taskCounter == 1) {
+            task = TASK_SINGLE;
+        }
+        System.out.println(PRINT_TASK_MESSAGE_FRONT + taskCounter + SPACING + task + PRINT_TASK_MESSAGE_BACK);
+    }
+
+    /**
+     * Scans for the description of the input task
+     *
+     * @param chatInput input of the user
+     * @return description of task
+     */
+    private static String scanDescription(String chatInput) {
+        String[] words = chatInput.split(SPACING);
+
+        int spaceCount = 1;
+
+        int startIdx = words[0].length() + spaceCount;
+        int endIdx = 0;
+
+        for (int i = 0; i < words.length; i++) {
+            if (words[i].charAt(0) == '/') {
+                break;
+            }
+            endIdx += words[i].length() + spaceCount;
+        }
+        return chatInput.substring(startIdx, endIdx - spaceCount);
+    }
+
+    /**
+     * Scans for the event time/due date of deadline of task
+     *
+     * @param chatInput input of user
+     * @return either the due date of deadline or event time
+     */
+    private static String scanTimePeriod(String chatInput) {
+        String[] words = chatInput.split(SPACING);
+
+        int spaceCount = 1;
+        int startIdx = 0;
+
+        for (int i = 0; i < words.length; i++) {
+            if (isAtEntry(words[i])) {
+                startIdx += ENTRY_AT.length();
+                break;
+            } else if (isByEntry(words[i])) {
+                startIdx += ENTRY_BY.length();
+                break;
+            }
+            startIdx += words[i].length() + spaceCount;
+        }
+        return chatInput.substring(startIdx + spaceCount);
+    }
+
+    /**
+     * Scans for the command (keyword)
+     *
+     * @param chatInput input of user
+     * @return the command to be executed
+     */
+    private static String scanKeyword(String chatInput) {
+        String[] words = chatInput.toLowerCase().split(SPACING);
+        return words[0];
+    }
+
+    /**
+     * Searches for the task number in the command
+     *
+     * @param chatInput input of user
+     * @return task number to be marked as done
+     */
+    private static int findTaskNumber(String chatInput) {
+        int accountForArray = 1;
+        String[] words = chatInput.split(SPACING);
+        int taskNumber = Integer.parseInt(words[1]) - accountForArray;
+        return taskNumber;
+    }
+
+    /**
+     * Checks if input string is /at
+     *
+     * @param input input of entry
+     * @return true if entry is /at
+     */
+    private static boolean isAtEntry(String input) {
```

Comment:
Perhaps the name of this method could be refactored to checkIfIsAtEntry since isAtEntry sounds more like the name of a boolean variable than a method name.